pkg/storage/stores/shipper/compactor/retention/retention_test.go

```diff
@@ -29,22 +30,27 @@ import (
 
 type mockChunkClient struct {
 	mtx           sync.Mutex
-	deletedChunks []string
+	deletedChunks map[string]struct{}
```

Comment:
This fixes a flaky test when there can be duplicate chunk ids due to them overlapping 2 days and hence getting indexed in 2 tables. When chunk from each table gets deleted they get added here, hence creating a duplicate entry.

Reply:
Awesome !